src/components/Dropdown/DropdownMenu.tsx

```diff
@@ -0,0 +1,46 @@
+'use client';
+
+import clsx from 'clsx';
+import React from 'react';
+
+import type { DropdownOption } from './Dropdown';
+
+interface DropdownMenuProps {
+  options: DropdownOption[];
+  onSelect: (option: DropdownOption) => void;
+  close: () => void;
+  size: 'sm' | 'md' | 'lg';
+}
+
+export function DropdownMenu({ options, onSelect, close, size }: DropdownMenuProps) {
+  const baseStyle = clsx(
+    'absolute block in-w-max text-gray-900 bg-white border my-1 rounded-md shadow-md border-none',
+    'max-h-40 overflow-y-auto',
+    'overflow-hidden whitespace-nowrap text-ellipsis'
+  );
+  const menuStyle = 'px-4 py-2 transition-colors hover:bg-gray-200 cursor-pointer';
+  const sizes = {
+    sm: 'px-3 py-1.5 text-sm',
+    md: 'px-4 py-2 text-base',
+    lg: 'px-5 py-3 text-lg',
+  };
+
+  const menuClasses = clsx(menuStyle, sizes[size]);
+
+  return (
+    <ul className={baseStyle}>
+      {options.map(option => (
+        <li
+          className={menuClasses}
+          key={option.value}
+          onClick={() => {
+            onSelect(option);
+            close();
```

Comment:
```suggestion
            onSelect(option);
```
```
on close()
```
현재 close()가 부모와 자식에서 모두 호출되어 동일 상태 업데이트가 두 번 일어납니다. 닫기 책임은 부모에만 두고 자식에서는 선택 이벤트만 전달하는 게 좋습니다.